File: chapter2/selection_sort.py
import numpy as np
import time

def selection_sort(tab: np.array) -> np.array: 
    """
    Sort an array using selection sort algorithm(O(n^2))
    :param tab : input array
    :return : the sorted array
    """
    for i in range(0, len(tab)-1):
        key = tab[i]
        index = i
        for j in range(i+1, len(tab)):
            if tab[j] < key:
                key = tab[j]
                index = j
        if index != i:
            temp = tab[i]
            tab[i] = tab[index]
            tab[index] = temp
    return tab

tab = np.array([5, 2, 4, 6, 1, 3])
start = time.time()
tab_sorted = selection_sort(tab)
end = time.time()
print(tab_sorted)  
print(f"Done in {end-start} ms")     

File: chapter2/selection_sort.js
function selection_sort(tab) {
    /*
        Function: selection_sort
        Sort an array using selection sort algorithm(O(n^2))
        Parameters:
            tab: input array
        Return:
            the sorted array
    */
   for(let i=0; i<tab.length-1; i++) {
        let key = tab[i]
        let index = i
        for(let j=i+1; j<tab.length; j++) {
            if(tab[j] < key) {
                key = tab[j]
                index = j
            }
        }
        if(index != i) {
            let temp = tab[i]
            tab[i] = tab[index]
            tab[index] = temp
        }
   }
   return tab
}

tab = [5, 2, 4, 6, 1, 3]
tab_sorted = selection_sort(tab)
console.log(tab_sorted)